VideoWeb/VideoWeb/ClientApp/src/app/app.component.ts

```diff
@@ -123,8 +163,22 @@ export class AppComponent implements OnInit, OnDestroy {
         );
     }
 
+    private setupSecurityServiceProviderSubscription() {
+        this.securityServiceProviderService.currentSecurityService$.pipe(takeUntil(this.destroyed$)).subscribe(service => {
+            this.securityService = service;
+            this.serviceChanged$.next();
+
+            service.isAuthenticated$
+                .pipe(takeUntil(this.serviceChanged$), takeUntil(this.destroyed$), delay(0)) // delay(0) pipe is to prevent angular ExpressionChangedAfterItHasBeenCheckedError
+                .subscribe(authenticated => {
+                    this.loggedIn = authenticated;
+                });
+        });
+    }
+
     ngOnDestroy(): void {
         this.subscriptions.unsubscribe();
+        this.destroyed$.next();
```

Comment:
It came up in a call about calling this.destroyed$.complete() after this call. I agreed at the time as I wasn't totally sure but after researching a bit further I have determined that this is not needed. As there are many other cases where we use subjects and rely on garbage collection to complete I will not include it.

https://stackoverflow.com/questions/57007118/do-i-need-to-complete-takeuntil-subject-inside-ngondestroy/57009988
https://stackoverflow.com/questions/44289859/do-i-need-to-complete-a-subject-for-it-to-be-garbage-collected